tests/foreman/api/test_operatingsystem.py

```diff
@@ -142,41 +137,43 @@ def test_positive_read_minor_version_as_string(self):
         """
         minor = int(gen_string('numeric', random.randint(1, 16)))
         operating_sys = entities.OperatingSystem(minor=minor).create()
-        self.assertEqual(operating_sys.minor, str(minor))
+        assert operating_sys.minor == str(minor)
 
     @pytest.mark.tier1
-    def test_positive_create_with_description(self):
+    @pytest.mark.parametrize('desc', **parametrized(valid_data_list()))
+    def test_positive_create_with_description(self, desc):
         """Create operating system with description
 
         :id: 980e6411-da11-4fec-ae46-47722367ae40
 
+        :parametrized: yes
+
         :expectedresults: Operating system entity is created and has proper
             description
 
         :CaseImportance: Critical
         """
         name = gen_string('utf8')
-        for desc in valid_data_list():
-            with self.subTest(desc):
-                os = entities.OperatingSystem(name=name, description=desc).create()
-                self.assertEqual(os.name, name)
-                self.assertEqual(os.description, desc)
+        os = entities.OperatingSystem(name=name, description=desc).create()
+        assert os.name == name
+        assert os.description == desc
```

Comment:
Add the cleanup.

Reply:
There is currently no cleanup done, so I'm not adding any as part of the pytest conversion. Based on discussion with @mshriver I am holding off on adding any cleanup until the team has discussed the pros and cons involved.